src/types/dict.rs

```diff
@@ -213,13 +213,13 @@ impl PyDict {
 
     /// Returns an iterator of `(key, value)` pairs in this dictionary.
     ///
-    /// Note that it's unsafe to use when the dictionary might be changed by
-    /// other code.
+    /// # Panics
+    ///
+    /// If PyO3 detects that the dictionary is mutated during iteration, it will panic.
+    /// It is allowed to modify the values of the keys as you iterate over the dictionary, but only
```

Comment:
I know "modify the values of the keys" is copied verbatim from the C-API docs, but I think it's quite confusing.